dpe/src/crypto.rs

```diff
@@ -44,8 +44,34 @@ impl Default for EcdsaPub {
     }
 }
 
+pub trait Hasher: Sized {
+    /// Adds a chunk to the running hash.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - Value to add to hash.
+    fn update(&mut self, bytes: &[u8]) -> Result<(), DpeErrorCode>;
+
+    /// Finish a running hash operation and return the result.
+    ///
+    /// Once this function has been called, the object can no longer be used and
+    /// a new one must be created to hash more data.
+    ///
+    /// # Arguments
+    ///
+    /// * `digest` - Where the computed digest should be written.
+    fn finish_mut(self, digest: &mut [u8]) -> Result<(), DpeErrorCode>;
+
+    fn finish(self) -> Result<[u8; DPE_PROFILE.get_hash_size()], DpeErrorCode> {
```

Comment:
Why do we assume `DPE_PROFILE` here and let the user choose the profile in `Crypto::hash()` and `Crypto::hash_initialize()`? It looks like we either want to always assume `DPE_PROFILE` or make `Hasher` depend on the profile, like `trait Hasher<const PROFILE: DpeProfile>` (we may need some nightly feature, not sure).

Reply:
That's a good point. I spaced we weren't depending on that definition anywhere else for `Crypto`. I removed the function.